### ISSUE TITLE
Question about prediction using scaling law

### ISSUE THREAD

**Issue by x54-729** [huggingface/datablations#9]
Dear authors, I am recently reading your amazing working of this paper, however I encounter a calculation problem while trying to predict loss.
For figure 5 left, I choose 8.7B parameters, 187B tokens and 0.14 unique tokens as an example and follow this program:

```python
import math

alpha = 0.3526596
beta = 0.3526596
A = math.exp(6.255414)
B = math.exp(7.3049974)
E = math.exp(0.6254804)

a = beta / (alpha + beta)
b = alpha / (alpha + beta)
G = pow((alpha * A) / (beta * B), 1/(alpha + beta))
# C = 9.3e21
# D_opt = (1 / G) * pow(C / 6, b)
# N_opt = G * pow(C / 6, a)
# print(f"D_opt: {'{:.2e}'.format(D_opt)}, N_opt: {'{:.2e}'.format(N_opt)}")

N = 8.7e9
D = 178e9
U_d = D / 7
U_n = min(pow(U_d*G, beta/alpha)*G, N)
R_n = max(N / U_n - 1, 0)
R_d = D / U_d - 1
R_nn = 5.309743 
R_dd = 15.387756
L = E + (A / pow(U_n + R_nn * U_n * (1 - math.exp(-R_n / R_nn)), alpha)) + (B / pow(U_d + U_d*R_dd*(1 - math.exp(-R_d / R_dd)), beta))
print(f"Predicted loss: {L}")
```

The final output is 2.237666982866588 which is not close to 2.38 reference to the figure. I think there must be something wrong with the paramters, but I can not find it out. Could you please tell me what's wrong with my understanding of the scaling law?

**Comment by Muennighoff** [comment]
The loss values predicted by the formula are unlikely to exactly match the final loss of the model. 

This is because we borrow some of its parameters from the values put forth in [Chinchilla](https://arxiv.org/abs/2203.15556), which were created using a different setup, some different hyperparameters (learning rate, schedule etc.) leading to different loss values (See Appendix B). However, the trend predicted by the formula is expected to be accurate, e.g. plugging in two model configurations to compare which one should have a better loss (as for Figure 1, right). Or if you e.g. modify `U_d = D / 7` to `U_d = D / 5` in your code, you will get a sense of how much your loss should improve relatively by having more unique data (e.g. relatively speaking `U_d = D / 7` -> `U_d = D / 5` gives a higher boost than `U_d = D / 3` -> `U_d = D / 1`).

In Figure 5 left, the loss predictions from the formula are all shifted by a constant, such that they match with the actual loss at 100%. The point of this Figure is to show how well the trend matches not the actual predicted loss values. Hence the caption says `Loss curves predicted by our data-constrained scaling laws are shifted to exactly match the loss at 100% unique data. `, but maybe this isn't made clear enough. Let me know if you have an idea for making this clearer in the paper / the repo! 🧐

**Comment by x54-729** [comment]
That's clear enough for me, thank you for your explanation!
